packages/core/src/Editor.ts

```diff
@@ -162,6 +165,29 @@ export class Editor extends EventEmitter<EditorEvents> {
     }, 0)
   }
 
+  /**
+   * Remove the editor from the DOM, but still allow remounting at a different point in time
+   */
+  public unmount() {
+    if (this.editorView) {
+      // Cleanup our reference to prevent circular references which caused memory leaks
+      // @ts-ignore
+      const dom = this.editorView.dom as TiptapEditorHTMLElement
+
+      if (dom && dom.editor) {
+        delete dom.editor
+      }
+      this.editorView.destroy()
+    }
+    this.editorView = null
+    this.isInitialized = false
+
+    if (this.css) {
+      this.css.remove()
```

Comment:
I think we could just use optional chaining here removing the unneeded if-wrapping.